commands/base_command.go

```diff
@@ -104,144 +100,91 @@ func (c *BaseCommand) Usage(message string) {
-func (c *BaseCommand) NewRestClient(host string) restclient.RestClientOperations {
-	return c.clientFactory.NewRestClient(host, c.transport, c.jar, c.tokenFactory)
-}
-
-// NewMtaClient creates a new MTA deployer REST client
-func (c *BaseCommand) NewMtaClient(host string) (mtaclient.MtaClientOperations, error) {
-	space, err := c.GetSpace()
+	parser := NewCommandFlagsParser(flags, c.flagsParser, c.flagsValidator)
+	err := parser.Parse(args)
 	if err != nil {
-		return nil, err
+		c.Usage(err.Error())
+		return Failure
 	}
 
-	return c.clientFactory.NewMtaClient(host, space.Guid, c.transport, c.jar, c.tokenFactory), nil
-}
+	deployServiceUrl, err := c.deployServiceURLCalculator.ComputeDeployServiceURL(GetStringOpt(deployServiceURLOpt, flags))
+	if err != nil {
+		ui.Failed(err.Error())
+		return Failure
+	}
 
-// NewMtaV2Client creates a new MTAV2 deployer REST client
-func (c *BaseCommand) NewMtaV2Client(host string) (mtaclient_v2.MtaV2ClientOperations, error) {
-	space, err := c.GetSpace()
+	cfTarget, err := c.GetCFTarget()
 	if err != nil {
-		return nil, err
+		ui.Failed(err.Error())
+		return Failure
 	}
 
-	return c.clientFactory.NewMtaV2Client(host, space.Guid, c.transport, c.jar, c.tokenFactory), nil
+	return c.executeInternal(parser.Args(), deployServiceUrl, flags, cfTarget)
 }
 
-// Context holding the username, Org and Space of the current used
-type Context struct {
-	Username string
-	Org      string
-	Space    string
+// GetBoolOpt gets the option identified by the specified name.
+func GetBoolOpt(name string, flags *flag.FlagSet) bool {
+	opt, _ := strconv.ParseBool(GetStringOpt(name, flags))
+	return opt
 }
 
-// GetContext initializes and retrieves the Context
-func (c *BaseCommand) GetContext() (Context, error) {
-	username, err := c.GetUsername()
-	if err != nil {
-		return Context{}, err
-	}
-	org, err := c.GetOrg()
-	if err != nil {
-		return Context{}, err
-	}
-	space, err := c.GetSpace()
-	if err != nil {
-		return Context{}, err
-	}
-	return Context{Org: org.Name, Space: space.Name, Username: username}, nil
+// GetStringOpt gets the option identified by the specified name.
+func GetStringOpt(name string, flags *flag.FlagSet) string {
+	return flags.Lookup(name).Value.String()
 }
 
-// GetOrg gets the current org name from the CLI connection
-func (c *BaseCommand) GetOrg() (plugin_models.Organization, error) {
-	org, err := c.cliConnection.GetCurrentOrg()
-	if err != nil {
-		return plugin_models.Organization{}, fmt.Errorf("Could not get current org: %s", err)
-	}
-	if org.Name == "" {
-		return plugin_models.Organization{}, fmt.Errorf("No org and space targeted, use '%s' to target an org and a space", terminal.CommandColor("cf target -o ORG -s SPACE"))
-	}
-	return org, nil
+// GetUintOpt gets the option identified by the specified name.
+func GetUintOpt(name string, flags *flag.FlagSet) uint {
+	opt, _ := strconv.ParseUint(GetStringOpt(name, flags), 0, strconv.IntSize)
```

Comment:
probably also good to check for the error?

Reply:
This won't return an error because if there is an invalid option value, the command will fail during flags parsing